pysmt/formula.py

```diff
@@ -897,14 +897,66 @@ def BVRepeat(self, formula, count=1):
         for _ in xrange(count-1):
             res = self.BVConcat(res, formula)
         return res
-
-    def Length(self, formula):
-        """ Creates an expression of the form:
-            Length formula
+    
+    def str_length(self, formula):
+        """ Creates an expression of the form Length formula
         """
-        n = self.create_node(node_type=op.LENGTH, args=(formula,))
+        t = self.env.stc.get_type(formula)
```

Comment:
No need to do this here. The typechecker is automatically called within ```create_node```. If the rule in the type checker is correct, then this check will be performed before leaving this function.